closed/get_j9_source.sh

```diff
@@ -30,144 +33,130 @@ usage() {
 	echo "  -omr-repo         the OpenJ9/omr repository url: https://github.com/eclipse/openj9-omr.git"
 	echo "                    or git@github.com:<namespace>/openj9-omr.git"
 	echo "  -omr-branch       the OpenJ9/omr git branch: openj9"
-	echo "  -omr-sha           a commit SHA for the omr repository"
+	echo "  -omr-sha          a commit SHA for the omr repository"
 	echo "  -parallel         (boolean) if 'true' then the clone j9 repository commands run in parallel, default is false"
 	echo ""
 	exit 1
 }
 
 # require bash 4.0 or later to support associative arrays
-bash_version=`bash --version | sed -n 1p`
-if [[ $bash_version != *"version 4."* ]] ; then
+if [ "0${BASH_VERSINFO[0]}" -lt 4 ] ; then
 	echo "Bash version 4.0 or later is required!"
 	exit 1
 fi
 
-declare -A j9repos
 declare -A branches
-declare -A default_j9repos=( [openj9]=eclipse/openj9 [omr]=eclipse/openj9-omr )
-declare -A default_branches=( [openj9]=master [omr]=openj9 )
 declare -A commands
+declare -A git_urls
 declare -A shas
 
-pflag="false"
-base_git_url=https://github.com
+git_urls[openj9]=https://github.com/eclipse/openj9
+branches[openj9]=master
+
+git_urls[omr]=https://github.com/eclipse/openj9-omr
+branches[omr]=openj9
+
+pflag=false
 
-for i in "$@"
-do
+for i in "$@" ; do
 	case $i in
 		-h | --help )
-		usage
-		;;
-
-		-r=* | --revision=* )
-		hgtag="${i#*=}"
-		;;
+			usage
+			;;
 
 		-openj9-repo=* )
-		j9repos[openj9]="${i#*=}"
-		;;
+			git_urls[openj9]="${i#*=}"
+			;;
 
 		-openj9-branch=* )
-		branches[openj9]="${i#*=}"
-		;;
+			branches[openj9]="${i#*=}"
+			;;
 
 		-openj9-sha=* )
-		shas[openj9]="${i#*=}"
-		;;
+			shas[openj9]="${i#*=}"
+			;;
 
 		-omr-repo=* )
-		j9repos[omr]="${i#*=}"
-		;;
+			git_urls[omr]="${i#*=}"
+			;;
 
 		-omr-branch=* )
-		branches[omr]="${i#*=}"
-		;;
+			branches[omr]="${i#*=}"
+			;;
 
 		-omr-sha=* )
-		shas[omr]="${i#*=}"
-		;;
+			shas[omr]="${i#*=}"
+			;;
 
 		-parallel=* )
-		pflag="${i#*=}"
-		;;
+			pflag="${i#*=}"
+			;;
 
 		'--' ) # no more options
-		usage
-		;;
+			break
+			;;
 
 		-*) # bad option
-		usage
-		;;
+			usage
+			;;
 
 		*) # bad option
-		usage
-		;;
+			usage
+			;;
 	esac
 done
 
-git=`which git`
-
 # clone OpenJ9 repos
 date '+[%F %T] Get OpenJ9 sources'
 START_TIME=$(date +%s)
 
-for i in "${!default_j9repos[@]}" ; do
-	branch=${default_branches[$i]}
-	if [ ${branches[$i]+_} ]; then
-		branch=${branches[$i]}
-	fi
+for i in "${!git_urls[@]}" ; do
+	branch=${branches[$i]}
 
-	if [ -d ${i} ]; then
+	if [ -d ${i} ] ; then
 		echo
 		echo "Update ${i} source"
 		echo
 
 		cd ${i}
-		git pull --rebase origin ${branch} || exit $?
+		git pull --rebase origin ${branch}
 
-		if [ -f .gitmodules ]; then
-			git pull --rebase --recurse-submodules=yes || exit $?
-			git submodule update --rebase --recursive || exit $?
+		if [ -f .gitmodules ] ; then
+			git pull --rebase --recurse-submodules=yes
```

Comment:
submodules aren't in use here.  should we remove any submodule reference and re-add in the future if there is a case where we are using a submodule?

Reply:
We could remove this, but I don't think its presence is causing any harm so I'm inclined to leave it.

Reply:
True, I'm fine with that.